[WIP] add UnitSeries neurodata_type

ajtritt opened the pull request:
This is a proposal/work-in-progress for addressing #194. It adds a new type, called *UnitSeries*, that would replace *Clustering*. It extends *TimeSeries* and requires that *TimeSeries.data* be an `int`. It also adds an additional dataset, called *units* that is a DynamicTableRegion that points to *Units* table.

ajtritt committed:
add UnitSeries neurodata_type

tjd2002 commented:
Would this allow two or more spikes from different units with the same timestamp? (This can happen in real data)

ajtritt commented:
@tjd2002 

> Would this allow two or more spikes from different units with the same timestamp? (This can happen in real data)

Yes, there are no restrictions on unique time stamps in *TimeSeries*

bendichter commented:
I'm fine with the name change, and I like that this now inherits from `TimeSeries`, but it would be nice if there was some kind of link to `SpikeEventSeries` (https://github.com/NeurodataWithoutBorders/nwb-schema/issues/239). As it stands now, if you had multiple `UnitSeries` and multiple `SpikeEventSeries` objects (e.g., 6, one for each of 6 shanks), it is unclear how you would indicate which `SpikeEventSeries` object is connected to which `UnitSeries` object. I guess you could do it implicitly if both had a field for referencing the regions of the `Units` table, and you implied that objects that referenced the same Units table region were connected.

bendichter committed:
add links:

ElectrodeGroup -> SpikeEventSeries
SpikeEventSeries -> UnitSeries
UnitSeries -> Units table

bendichter committed:
add optional electrode_group link to ElectricalSeries and make electrodes optional

rly committed:
Merge branch 'dev' into enh/unit_time_series

rly commented:
I believe this is covered by the Units table with its waveforms column and link to the electrodes table. Please re-open if not.
